engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/mutable/SortNode.java

```diff
@@ -52,19 +67,30 @@ Object doOther(State state, Object self, Object comparator) {
         this);
   }
 
-  Object runSort(Comparator<Object> compare, Array self, EnsoContext context) {
-    doSort(self.getItems(), compare);
-    LoopNode.reportLoopCount(this, (int) self.length());
+  Object runSort(Comparator<Object> compare, Object[] self, EnsoContext context) {
+    doSort(self, compare);
+    LoopNode.reportLoopCount(this, self.length);
     return context.getBuiltins().nothing();
   }
 
+  Object runSort(Comparator<Object> compare, byte[] self, EnsoContext context) {
+    Byte[] tmpArray = ArrayUtils.toObject(self);
```

Comment:
> Because one cannot use a custom comparator with primitive (Java) arrays there is a conversion needed.

This kind of conversion isn't going to be fast. I doubt the compiler will be able to virtualize away the `Byte[]` allocation. Anyway the sorting is done by `doSort` method which is behind `@TruffleBoundary`, so we can just dream of PE performance. Moreover we are unlikely to do anything better in this case (unless we want to write our own sort). 

PS: I was thinking of creating a _virtualized_ `java.util.List` delegating to the byte buffer and then just calling `List.sort()`, but the implementation in the `default List.sort` isn't really helping us anyway. The first thing it does is `toArray()` conversion.

Reply:
Yes, performance wasn't the biggest priority at this point, compatibility was.
So far we don't have any other use-cases where this is really needed so I can file a ticket to investigate it although it won't be a priority.